vocabulary-care-pathways/api/src/main/java/org/phenotips/vocabulary/internal/solr/CPNode.java

```diff
@@ -0,0 +1,193 @@
+     */
+    CPNode(@Nonnull final String id, @Nonnull final String name)
+    {
+        this.id = id;
+        this.name = name;
+        this.isA = new HashSet<>();
+        this.termCategory = new HashSet<>();
+    }
+
+    /**
+     * Gets the node {@link #id}.
+     *
+     * @return the node identifier
+     */
+    @Nonnull
+    String getId()
+    {
+        return this.id;
+    }
+
+    /**
+     * Gets the node {@link #name}.
+     *
+     * @return the node name
+     */
+    @Nonnull
+    String getName()
+    {
+        return this.name;
+    }
+
+    /**
+     * Returns the collection of parent identifiers for the node.
+     *
+     * @return a collection of parents
+     */
+    @Nonnull
+    Collection<String> getParents()
+    {
+        return Collections.unmodifiableSet(this.isA);
+    }
+
+    /**
+     * Returns the collection of ancestor identifiers for the node.
+     *
+     * @return a collection of identifiers
+     */
+    @Nonnull
+    Collection<String> getAncestors()
+    {
+        return Collections.unmodifiableSet(this.termCategory);
+    }
+
+    /**
+     * Adds a parent to the collection of {@link #isA parents}, and returns the updated collection.
+     *
+     * @param parent the parent identifier to add
+     * @return the updated collection of parent identifiers
+     */
+    @Nonnull
+    Collection<String> addParent(@Nullable final String parent)
+    {
+        CollectionUtils.addIgnoreNull(this.isA, parent);
+        return Collections.unmodifiableSet(this.isA);
+    }
+
+    /**
+     * Adds an ancestor to the collection of {@link #termCategory ancestors}, and returns the updated collection.
+     *
+     * @param ancestor the ancestor identifier to add
+     * @return the updated collection of ancestor identifiers
+     */
+    @Nonnull
+    Collection<String> addAncestor(@Nullable final String ancestor)
+    {
+        CollectionUtils.addIgnoreNull(this.termCategory, ancestor);
+        return Collections.unmodifiableSet(this.termCategory);
+    }
+
+    /**
+     * Adds a parents to the collection of {@link #isA parents}, and returns the updated collection.
+     *
+     * @param parents the collection of parent identifiers to add
+     * @return the updated collection of parent identifiers
+     */
+    @Nonnull
+    Collection<String> addParents(@Nullable final Collection<String> parents)
+    {
+        if (CollectionUtils.isNotEmpty(parents)) {
+            this.isA.addAll(parents);
+        }
+        return Collections.unmodifiableSet(this.isA);
+    }
+
+    /**
+     * Replaces the collection of {@link #isA parents} with {@code parents}, and returns the updated collection.
+     *
+     * @param parents the new collection of parent identifiers
+     * @return the updated collection of parent identifiers
+     */
+    @Nonnull
+    Collection<String> setParents(@Nullable final Set<String> parents)
+    {
+        if (parents != null) {
+            this.isA = parents;
```

Comment:
The value may be readonly, which would cause further calls to add parents to fail. To be safe, use a new set built from the passed value?